Cue/app/tabs/library/LibraryHome.js

```diff
@@ -99,18 +133,121 @@ class LibraryHome extends React.Component {
+
+      // Trigger a sync if needsSync is set or if there are any pending localChanges
+      if (this.state.needsSync || this.props.localChanges.length) {
+        if (this.state.connected) {
+          if (!this._shouldThrottleAutoSync()) {
+            console.info('_onNavigatorEvent: Sync is pending and network is available; refreshing now '
+              + `(needsSync: ${this.state.needsSync.toString()}, `
+              + `localChanges.length: ${this.props.localChanges.length.toString()})`)
+
+            this._refresh()
+          } else {
+            console.info('_onNavigatorEvent: Sync is pending and network is available but sync is being throttled; doing nothing '
+              + `(needsSync: ${this.state.needsSync.toString()}, `
+              + `localChanges.length: ${this.props.localChanges.length.toString()})`)
+          }
+        } else {
+          console.info('_onNavigatorEvent: Sync is pending but network is not available; doing nothing '
+            + `(needsSync: ${this.state.needsSync.toString()}, `
+            + `localChanges.length: ${this.props.localChanges.length.toString()})`)
+        }
+      } else {
+        console.info(`_onNavigatorEvent: No sync pending `
+          + `(needsSync: ${this.state.needsSync.toString()}, `
+          + `localChanges.length: ${this.props.localChanges.length.toString()})`)
+      }
+    }
+  }
+
+  _onNetworkIsConnectedChanged = (isConnected: boolean) => {
+    if (isConnected !== this.state.connected) {
+      console.info(`_onNetworkIsConnectedChanged: Network status changed: `
+        + `${(isConnected ? 'connected' : 'not connected')}`)
+
+      this.setState({connected: isConnected})
+
+      let shouldSync = isConnected && !this._shouldThrottleAutoSync()
+
+      if (shouldSync) {
+        console.info('_onNetworkIsConnectedChanged: Requesting sync due to network status change')
+
+        if (this._isLibraryHomeInForeground()) {
+          console.info('_onNetworkIsConnectedChanged: '
+            + 'LibraryHome is in the foreground; refreshing now')
+
+          this._refresh()
+        } else {
+          console.info('_onNetworkIsConnectedChanged: '
+            + 'LibraryHome is not in the foreground, setting needsSync')
+
+          this.setState({needsSync: true})
+        }
+      } else {
+        console.info(`_onNetworkIsConnectedChanged: `
+          + `Not triggering sync (isConnected: ${isConnected.toString()})`)
+      }
+    }
   }
 
   _refresh = () => {
     if (!this.state.refreshing) {
       this.setState({refreshing: true})
       this.props.onSyncLibrary(this.props.localChanges).then(failedSyncs =>{
         if (failedSyncs && failedSyncs.length) {
-          this.setState({refreshing: false})
+          this.setState({refreshing: false, lastSyncTime: new Date(), needsSync: false})
           this.props.navigator.push({failedSyncs})
         } else {
           this.props.onLoadLibrary().then(response => {
```

Comment:
When I have no connection the error gets thrown here it is not being caught by the catch at line 254.

This is causing the infinite spinning bug to occur for me and no alert